### PR TITLE
feat: update deposit address to selected created explicit account

### LINKED ISSUE #7846: [Task]: Update deposit address to selected created explicit account
### Task description

Once the account is created, the selected account to be used for every account centric action, will also determine the deposit address

### PR DESCRIPTION
**Pull request by evavirseda**
## Summary

> Closes #7846 

...

## Changelog

```
- Please write a list of changes
```

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [x] Linux
    -   [ ] Windows

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [ ] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/lib/core/wallet/utils/getDepositAddress.ts [hunk 1]
```diff
@@ -1,6 +1,16 @@
...
 
-// TODO(2.0): the implementation is incorrect, the deposit address should be the address of the selected block issuer account output , not the ed25519 address
-export function getDepositAddress(wallet: IWallet): Promise<string> {
-    return wallet.address()
+// TODO(2.0): Update the implementation to handle multiple accounts and use the selected account's address.
+export async function getDepositAddress(): Promise<string> {
+    const wallet = get(selectedWallet)
+    const accounts = await wallet?.accounts()
+    if (accounts?.length) {
+        // TODO: when we have selectedAccount in the store,  we need to find the account with the blockIssuer feature
```
Comment (cpl121) [inline packages/shared/lib/core/wallet/utils/getDepositAddress.ts:10]: Duplicate TODO comment

File: packages/shared/lib/core/wallet/utils/getDepositAddress.ts [hunk 2]
```diff
@@ -1,6 +1,16 @@
...
+        const depositAddress = (accounts[0]?.output as AccountOutput).accountId ?? ''
+        return depositAddress
```
Comment (cpl121) [inline packages/shared/lib/core/wallet/utils/getDepositAddress.ts:12]:
```suggestion
        return (accounts[0]?.output as AccountOutput).accountId ?? ''
```

**Review by cpl121 — CHANGES_REQUESTED** [review]

**Review by cpl121 — APPROVED** [review]
Perfect!

File: packages/shared/lib/core/wallet/actions/buildWalletState.ts [hunk 3]
```diff
@@ -38,7 +38,7 @@ export async function buildWalletState(
 
     try {
         balances = await wallet.getBalance()
-        depositAddress = await getDepositAddress(wallet)
+        depositAddress = await getDepositAddress()
```
Comment (begonaalvarezd) [inline packages/shared/lib/core/wallet/actions/buildWalletState.ts:41]: lets not remove this argument so we leave the function a bit more generic

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]
🙏🏼 little thing

**Pull request merged**